apps/mark-scan/accessible-controller/src/controllerd.rs

```diff
@@ -206,29 +207,39 @@ fn send_key(device: &mut Device, key: keyboard::Key) -> Result<(), CommandError>
 fn handle_command(device: &mut Device, data: &[u8]) -> Result<(), CommandError> {
     let ButtonStatusCommand { button, action } = data.try_into()?;
 
-    let key: keyboard::Key;
+    let key: Keyboard;
     match action {
         Action::Pressed => match button {
             Button::Select => {
-                key = keyboard::Key::Enter;
+                key = uinput::event::Keyboard::Key(keyboard::Key::Enter);
             }
             Button::Left => {
-                key = keyboard::Key::Left;
+                key = uinput::event::Keyboard::Key(keyboard::Key::Left);
             }
             Button::Right => {
-                key = keyboard::Key::Right;
+                key = uinput::event::Keyboard::Key(keyboard::Key::Right);
             }
             Button::Up => {
-                key = keyboard::Key::Up;
+                key = uinput::event::Keyboard::Key(keyboard::Key::Up);
             }
             Button::Down => {
-                key = keyboard::Key::Down;
+                key = uinput::event::Keyboard::Key(keyboard::Key::Down);
+            }
+            Button::Help => key = uinput::event::Keyboard::Misc(keyboard::Misc::Help),
+            Button::RateDown => {
+                key = uinput::event::Keyboard::Key(keyboard::Key::LeftBrace);
+            }
+            Button::RateUp => {
+                key = uinput::event::Keyboard::Key(keyboard::Key::RightBrace);
             }
-            Button::Help => {
-                key = keyboard::Key::Q;
+            Button::VolumeDown => {
+                key = uinput::event::Keyboard::Misc(keyboard::Misc::VolumeDown);
             }
-            _ => {
-                return Err(CommandError::UnsupportedButton(button));
+            Button::VolumeUp => {
+                key = uinput::event::Keyboard::Misc(keyboard::Misc::VolumeUp);
+            }
```

Comment:
@kofi-q the 2 handlers for Volume Up/Down are not actually being called because the Volume Up/Down keys are being handled by the OS. IMO this is potentially a feature if it means we can avoid implementing Volume Up/Down hooks in code. But if the hooks do something more sophisticated than just volume up/down, or if this will cause confusion because the implementation differs from VxMark, then I can change they key codes to avoid changing volume at the system level. Thoughts?

Reply:
Ah, interesting! Also feels like a feature to me -- it could simplify things a bit, because we can then set the default OS volume directly to somewhere around midpoint and just have the controller increment/decrement as needed, rather than amplifying the raw signal from the screen reader audio.

We'd still have to do a bit of work either way to ensure that:
- we can reset the volume to the default level after every voter session
- the volume up/down buttons change the volume in steps of `< 10 db`
- we can clamp the volume to a minimum and maximum in the range specified in VVSG 2

Reply:
Doesn't point 3 mean we shouldn't send the volume up/down events directly from the controller daemon? It seems like that approach doesn't give us enough control to clamp.

Also realizing that this may not work outside GNOME - can verify if this works in OpenBox tomorrow.

Reply:
> Doesn't point 3 mean we shouldn't send the volume up/down events directly from the controller daemon?

Hm, at the time, I was thinking we might be able to set a minimum and maximum at the system level, but that might not actually be as straightforward as I'd hoped. If that's the case, might be best to assign to the volume controls to regular keyboard keys that we handle within the Vx apps